src/Compiler/Checking/CheckRecordSyntaxHelpers.fs

```diff
@@ -90,14 +91,24 @@ let TransformAstForNestedUpdates (cenv: TcFileState) (env: TcEnv) overallTy (lid
         let totalRange (origId: Ident) (id: Ident) =
             withStartEnd origId.idRange.End id.idRange.Start origId.idRange
 
-        match withExpr with
-        | SynExpr.Ident origId, (blockSep: BlockSeparator) ->
-            let lid, rng = upToId blockSep.Range id (origId :: ids)
+        let rangeOfBlockSeparator (id: Ident) =
+            let idEnd = id.idRange.End
+            let blockSeparatorStartCol = idEnd.Column
+            let blockSeparatorEndCol = blockSeparatorStartCol + 4
```

Comment:
Where does `4` magic constant come from? If we assume that a separator is `;` or `,` then it can't be 4 characters long.

If this is a `with` keyword range, it should not be calculated from the id, it's job of the parser to preserve it correctly, like it was before.